src/main/java/com/scalar/cassy/transferer/BackupPath.java

```diff
@@ -20,16 +20,24 @@ public static String create(BackupConfig config, String path) {
   }
 
   public static String create(RestoreConfig config, String path) {
-    String type = NODE_BACKUP_KEY;
-    if (config.getRestoreType().equals(RestoreType.CLUSTER)) {
-      type = CLUSTER_BACKUP_KEY;
-    }
-    return create(type, config, path);
+    return create(getType(config), config, path);
+  }
+
+  public static String createRoot(RestoreConfig config) {
+    return getType(config);
   }
 
   private static String create(String type, BaseConfig config, String path) {
     return Paths.get(
             type, config.getClusterId(), config.getSnapshotId(), config.getTargetIp(), path)
         .toString();
   }
+
+  private static String getType(RestoreConfig config) {
```

Comment:
It would be great if we can make this method like this and public so that we don't need to have `createRoot` method.  Using `instanceof` is not great though ... 
```
  private static String getType(BaseConfig config) {
    String type = NODE_BACKUP_KEY;
    if (config instanceof BackupConfig) {
      // ...
    }
    if (config instanceof RestoreConfig) {
      // ...
    }
    return type;
  }
```

Reply:
Sure, I guess using `instanceof` is the only way if we want to keep the `BackupPath` generation logic separated from the `BackupConfig` or `RestoreConfig` classes.